### ISSUE TITLE
" SyntaxError: Unexpected token { "

### ISSUE THREAD

**Issue by yourse1f-yourorg** [laggingreflex/npm-link-save#2]
This looked like exactly what I needed, but : 

    yourself@yours:~/projects/mmks$ npm-link-save mmks_widget
    /home/yourself/.npm-global/lib/node_modules/npm-link-save/index.js:4
    const { spawn } = require('child_process');
          ^

    SyntaxError: Unexpected token {
        at exports.runInThisContext (vm.js:53:16)
        at Module._compile (module.js:373:25)
        at Object.Module._extensions..js (module.js:416:10)
        at Module.load (module.js:343:32)
        at Function.Module._load (module.js:300:12)
        at Function.Module.runMain (module.js:441:10)
        at startup (node.js:139:18)
        at node.js:990:3
    yourself@yours:~/projects/mmks$ npm --version
    2.15.11
    yourself@yours:~/projects/mmks$ node --version
    v4.7.0

**Comment by laggingreflex** [comment]
It uses ES2016 features that your node version (4.7) doesn't support yet. You need I think node 6+ (I'd just go with latest 7.9). 

Although it wouldn't be much trouble for me to support previous versions, it's only some handful syntaxes that will need to be changed. Let me know if keeping your current node version is a hard requirement for you and I'll see to it.

**Comment by yourse1f-yourorg** [comment]
I'm using Meteor which has been stuck at 4 for too long.  6 is on the road map, but not for some months yet.
Btw, this is a little gem of a tool. Thanks for doing it.

**Comment by laggingreflex** [comment]
I'm not too familiar with Meteor, but isn't it just a framework? What does a Node version has to do with it? Especially a higher version node which is backwards compatible. 

I think Meteor's compatibity issues is more that.. you won't be able to use ES2015 features in your project, which is ok, don't use them and it'll work fine. At least that's my current understanding, let me know if it helps or if I'm mistaken

**Comment by laggingreflex** [comment]
Whoops, sorry my bad. Didn't know it shipped with its own executable and everything. 

I'll see what I can do.

**Comment by yourse1f-yourorg** [comment]
No. Meteor depends on Node 4, but is using ES2015 for everything else.

Their deep integration into Node 4 has made it a big job to upgrade, ES2015 had to come first.  The trouble is this : 

I've been running ...

     meteor npm link mmks_widget

... but this fails ...

     meteor npm-link-save mmks_widget

... with `command not known`.

If I try ...

     npm-link-save mmks_widget

... then it uses Meteor's `npm` with none of Meteor's integrations -- or something like that.

In short, downgrading `npm-link-save` would be a help.
Specifying a minimum required `npm` version would be a courtesy to other future visitors here.

**Comment by laggingreflex** [comment]
Would you be able to take a shot at it? It would be a great help. I'm not too sure which features are present in node v4 and which aren't. I usually use latest features and have a babel build step. But I wanna avoid babel for this as it's just a single file.

**Comment by yourse1f-yourorg** [comment]
That's a little too far outside of my experience.  Plus, I'm fighting fires with other stuff right now.
It's the kind of thing I want to learn, but I'm spending way too much time learning at the cost of time producing and, er ... earning. :-(

**Comment by laggingreflex** [comment]
Fixed in https://github.com/laggingreflex/npm-link-save/commit/01e53e3e282171cd5e955b1e497923eba3ecf5a9, released v0.4.2. Check it out and let me know

**Comment by yourse1f-yourorg** [comment]
Oh GOD! I have three modules, practically identical as far as `npm` would look into them.

`npm-link-save` works for two out three! WTF!.

     npm ERR! 404 'mmks_book' is not in the npm registry.

and yes, the other two end up listed correctly in `package.json`.

**Comment by laggingreflex** [comment]
You probably forgot to do `npm link` first in `mmks_book` (I'm guessing it's a local module that you haven't published yet, hence 404)

**Comment by yourse1f-yourorg** [comment]
OK.  Got it figured out. Phew!

In each, module I was running . . . 

    meteor npm link;

. . . which puts a `symlink` somewhere deep in the bowels of Meteor.

If I simply run . . . 

    npm link;

. . . then the `symlink` goes in . . . 

    ~/.npm-global/lib/node_modules

When I checked just now, there were residual `symlinks` of the two modules that were being handled correctly.  So, executing in the project directory . . . 

    meteor npm link mmks_widget;  #  Success
    meteor npm link mmks_book;     #  Success
    meteor npm link mmks_layout;   #  Success

. .  but ...

    npm link mmks_widget;  #  Success
    npm link mmks_book;    #  Fail
    npm link mmks_layout;  #  Success

Probably not of any interest to you, @laggingreflex , but might give some future Googler a helpful nudge.
